src/timezones/Olson.jl

```diff
@@ -332,7 +341,10 @@ function resolve!(zone_name::AbstractString, zoneset::ZoneDict, ruleset::RuleDic
             end
         else
             if !haskey(ordered, rule_name)
-                ordered[rule_name] = order_rules(ruleset[rule_name]; max_year=max_year)
+                dates, ordered_rules, was_truncated = order_rules(ruleset[rule_name]; max_year=max_year)
+
+                ordered[rule_name] = dates, ordered_rules
+                truncated |= was_truncated
```

Comment:
The problem comes from ordered being used between calls to `resolve!`. See the [`resolve`](https://github.com/invenia/TimeZones.jl/blob/4650735488cb10734883bfa9182fadaa68533a4b/src/timezones/Olson.jl#L460) functions.